### ISSUE TITLE
🐛 Bug: Unintentional behaviour of the code snippet section 

### ISSUE THREAD

**Issue by AgniveshChaubey** [json-schema-org/website#417]
### Schema code snippet
When copying the schema using the `copy to clipboard button`, the title of that section in the top right corner changes to `data` (from `schema`) for a moment. I'm not sure if it is intentional, but it should not change I guess.

![image](https://github.com/json-schema-org/website/assets/113545196/42332920-eb3e-4892-a10a-16aae3576326)

After copying:

![image](https://github.com/json-schema-org/website/assets/113545196/5bae1e8a-3f27-4173-a365-8c2b42385d96)

---
### Instance code snippet
When copying the instance using the `copy to clipboard` button, the bottom row showing the validation result disappears for a moment, which also should not happen.
![image](https://github.com/json-schema-org/website/assets/113545196/d0fb9b51-d1e0-4cb9-8c7e-14141fd18374)

After copying:
![image](https://github.com/json-schema-org/website/assets/113545196/ae9ebae1-e802-422a-8f1a-4c90c87bdc3b)

### Expected Behavior

1. `schema` should not change to `data` when copying the schema snippet.
2. The bottom row should not disappear when copying the Instance snippet.

**Comment by DhairyaMajmudar** [comment]
If approved , I would like to take this up.

**Comment by benjagm** [comment]
Thanks Agnivesh!! @DhairyaMajmudar please go ahead.

**Comment by DhairyaMajmudar** [comment]
@benjagm @AgniveshChaubey I think the schema changing to data is an intentional behavior since while analysing the [code file](https://github.com/json-schema-org/website/blob/main/components/JsonEditor.tsx#L288)  I found that both of them are wrapped in a ternary conditional block depending on variable `isJsonSchema`

Pls. give your views on this.

**Comment by benjagm** [comment]
I think the label should stay consistent after copying data.

**Comment by AgniveshChaubey** [comment]
> I think the schema changing to data is an intentional behavior since while analysing the [code file](https://github.com/json-schema-org/website/blob/main/components/JsonEditor.tsx#L288) I found that both of them are wrapped in a ternary conditional block depending on variable isJsonSchema

I haven't gone through the codebase of the website, but when quickly looking at the file you mentioned, it seems that this ternary operator is  to check whether the snippet is a `JSON Schema` or a `JSON data`

```html
          <div className='flex flex-row items-center text-white h-6 font-sans bg-white/20 text-xs px-3 rounded-bl-lg font-semibold'>
            {isJsonSchema ? (
              <>
                <img src='/logo-white.svg' className='h-4 mr-1.5' /> schema
              </>
            ) : (
              <>data</>
            )}
          </div>
```

**Comment by benjagm** [comment]
Dhairya are you working on this?

**Comment by DhairyaMajmudar** [comment]
Yes I am trying to find its solution its taking time because its very anomalous , sometime the bug appears and sometimes not.

**Comment by DarhkVoyd** [comment]
@DhairyaMajmudar I am not sure but I believe it has something to do with how we are implementing the copy to clipboard component. Currently, instead of being a separate component with its own state, we are setting the copied state on the entire JsonEditor component due to which after each copy the Json Editor is caused to re-render again and again multiple times simply due to a copy event. Since, its somewhat heavy component with calculations, react would schedule updates and they would be async and in no particular order in some cases the state change would re-render but the data might not be ready and therefore it breaks to dynamically assert if the data is `schema` in some renders and `valid` or `invalid` in others, causing these errors.

**Comment by benjagm** [comment]
Thanks everyone! Great inputs here. I think the path forward is creating a new issue to review and refine the current code editor strategy. To do it more consistently.

**Comment by benjagm** [comment]
I am putting this on hold in favour of #560
